model/raw.proto

```diff
@@ -5,14 +5,41 @@ package model;
 import "commons.proto";
 
 message RawAccelerationData {
-  float sample_rate = 1;
+  float sample_rate_hz = 1;
   uint32 sample_count = 2;
-  repeated float acceleration_x = 3;
-  repeated float acceleration_y = 4;
-  repeated float acceleration_z = 5;
+  repeated float acceleration_x_g = 3;
+  repeated float acceleration_y_g = 4;
+  repeated float acceleration_z_g = 5;
+}
+
+/*
+Format containing the unscaled data from the sensor.
+
+This is a more compact format since protobuf has no built-in
+type for half-precision (16-bit) floats.
+
+The data is in little-endian.
+
+Formula for conversion:
+
+acceleration_g = 2 * range * acceleration_<x/y/z> / 2^16
+ */
+message RawAccelerationDataCompact {
+  float sample_rate_hz = 1;
+  uint32 sample_count = 2;
+  uint32 range = 3;
+  // Acceleration values contain 2 * sample_count bytes
+  bytes acceleration_x = 4;
```

Comment:
I suggest making these explicitly optional. An empty array or bytestring is not the same as the absence of one.

Reply:
I'm not against making them optional, but do you see a possible use in being able to differentiate empty arrays from non-existent ones?